src/test/scala/chisel3/tests/QueueTest.scala

```diff
@@ -57,4 +59,28 @@ class QueueTest extends FlatSpec with ChiselScalatestTester {
       }.join()
     }
   }
+
+  it should "work with IrrevocableIO" in{
+    test(new Module{
+      val io = IO(new Bundle{
+        val in = Flipped(Irrevocable(UInt(8.W)))
+        val out = Irrevocable(UInt(8.W))
+      })
+      val full = RegInit(false.B)
+      when(io.in.fire()){ full := true.B }.elsewhen(io.out.fire()){ full := false.B }
+      io.in.ready := !full
+      io.out.valid := full
+      io.out.bits := RegNext(io.in.bits)
```

Comment:
Would it make more sense to just instantiate a queue here, instead of essentially rolling your own one-element queue which doesn't appear to be fully correct (the data bits updates every cycle, regardless of io.in.ready)?

Reply:
The original purpose of instantiate a different queue is to distinguish between `DecoupledIO` and `IrrevocableIO`. Since there are additional requirements on signaling of `valid` and `ready`. I looked into the queue object and it satisfy the requirement. 
Fixed in  49d2857